app/controllers/my-projects/assignment/recommendations/add.js

```diff
@@ -257,49 +270,83 @@ export default class MyProjectsProjectRecommendationsAddController extends Contr
-    this.dispositionsChangesets.forEach(function(dispositionChangeset) {
-      dispositionChangeset.execute();
-    });
-    // statuscode = "Submitted" signifies in database that user has submitted recommendation
-    this.dispositions.setEach('statuscode', 'Submitted');
+    // array of true/false values each representing succesful upload of files to a disposition
+    const uploadResults = [];
+
+    try {
+      // loop through dispositions, uploading all files to each one
+      for (let i = 0; i < this.dispositions.length; i += 1) {
+        const disposition = this.dispositions.objectAt(i);
+
+        const fileUploadPromises = this.recommendationAddQueue.files.map(file => file.upload(`${ENV.host}/document`, {
+          fileKey: 'file',
+          data: {
+            instanceId: disposition.id,
+            entityName: 'dcp_communityboarddisposition',
+          },
+        }));
+
+        // await here because ember-file-upload requires a single file's upload() to complete before the
+        // next call of upload() on that file.
+        const fileUploadResponses = await Promise.all(fileUploadPromises); // eslint-disable-line
+
+        const filesUploadedToDispo = fileUploadResponses.every(res => res.status === 200);
 
-    this.dispositions.forEach((disposition) => {
-      if (this.allActions) {
-        disposition.set(targetField, this.dispositionForAllActions.recommendation);
+        uploadResults.push(filesUploadedToDispo);
+      }
+    } catch (e) {
+      this.set('submitError', true);
+      console.log(e); // eslint-disable-line
+    }
+
+    // Only proceed if all files were uploaded to all dispositions
+    if (uploadResults.every(res => res === true) && !this.submitError) {
+      const { participantType } = this;
+      const targetField = RECOMMENDATION_FIELD_BY_PARTICIPANT_TYPE_LOOKUP[participantType];
+      this.dispositionForAllActionsChangeset.execute();
+      this.dispositionsChangesets.forEach(function(dispositionChangeset) {
+        dispositionChangeset.execute();
+      });
+      // statuscode = "Submitted" signifies in database that user has submitted recommendation
+      this.dispositions.setEach('statuscode', 'Submitted');
+
+      this.dispositions.forEach((disposition) => {
+        if (this.allActions) {
+          disposition.set(targetField, this.dispositionForAllActions.recommendation);
+          disposition.setProperties({
+            dcpVotinginfavorrecommendation: this.dispositionForAllActions.dcpVotinginfavorrecommendation,
+            dcpVotingagainstrecommendation: this.dispositionForAllActions.dcpVotingagainstrecommendation,
+            dcpVotingabstainingonrecommendation: this.dispositionForAllActions.dcpVotingabstainingonrecommendation,
+            dcpTotalmembersappointedtotheboard: this.dispositionForAllActions.dcpTotalmembersappointedtotheboard,
+            dcpConsideration: this.dispositionForAllActions.dcpConsideration,
+          });
```

Comment:
You can make this a little easier to grok by using destructuring and object assignment shorthand:

const { a,b,c,d } = this.dispositionForAllActions;

disposition.setProperties({
a,b,c,d
});